locator/token_metadata.cc

```diff
@@ -95,9 +95,9 @@ class token_metadata_impl final {
     struct shallow_copy {};
 public:
     token_metadata_impl(shallow_copy, const token_metadata_impl& o) noexcept
-        : _topology(topology::config{})
+        : _topology(topology::config{}, topology::key_kind::inet_address)
```

Comment:
Please add contents to the commit message (`locator/topology: add key_kind parameter`)
about the motivation and plans for this feature - is it transient, only for the purpose of transitioning token_metadata to using host_id or is it meant to stay?

Reply:
> Please add contents to the commit message

Sure, I'll add the proper commit messages later, I just skipped them for this early highly experimental stage.

> about the motivation and plans for this feature - is it transient, only for the purpose of transitioning token_metadata to using host_id or is it meant to stay?

It's solely for the transitioning, when this PR is finished the key kind would be `key_kind::host_id` for everything and I'll just drop it.